lib/model/unittest/CMetricModelTest.cc

```diff
@@ -2292,26 +2292,32 @@ BOOST_FIXTURE_TEST_CASE(testIgnoreSamplingGivenDetectionRules, CTestFixture) {
     endTime += bucketLength;
     BOOST_REQUIRE_EQUAL(modelWithSkip->checksum(), modelNoSkip->checksum());
 
-    // this sample will be skipped by the detection rule
+    // Add data to both models
+    // the model with the detection rule will apply a small weighting to the sample
     for (std::size_t i = 0; i < 60; i++) {
+        this->addArrival(SMessage(startTime + i, "p1", 110.0), gathererNoSkip);
         this->addArrival(SMessage(startTime + i, "p1", 110.0), gathererWithSkip);
     }
+    modelNoSkip->sample(startTime, endTime, m_ResourceMonitor);
     modelWithSkip->sample(startTime, endTime, m_ResourceMonitor);
 
+    // Checksums will be different due to the small weighting applied to the sample
+    // added to the model with the detector rule.
+    BOOST_TEST_REQUIRE(modelWithSkip->checksum() != modelNoSkip->checksum());
+
     startTime = endTime;
     endTime += bucketLength;
 
-    // Wind the other model forward
-    modelNoSkip->skipSampling(startTime);
-
+    // Add more data to both models, for which the detection rule will not apply
     for (std::size_t i = 0; i < 60; i++) {
         this->addArrival(SMessage(startTime + i, "p1", 2.0), gathererNoSkip);
         this->addArrival(SMessage(startTime + i, "p1", 2.0), gathererWithSkip);
     }
     modelNoSkip->sample(startTime, endTime, m_ResourceMonitor);
     modelWithSkip->sample(startTime, endTime, m_ResourceMonitor);
 
-    // Checksums will be different due to the data gatherers
+    // Checksums will be different due to the small weighting applied to the sample
+    // added to the model with the detector rule.
     BOOST_TEST_REQUIRE(modelWithSkip->checksum() != modelNoSkip->checksum());
 
     // TODO this test fails due a different checksums for the decay rate and prior
```

Comment:
It would be good to replace the commented out code with some checks on the "underlying models" which are supposed to be the same.  If it's non-trivial then feel free to do this in a followup PR.

Reply:
The comment was out of date and misleading.  We now do expect the underlying models to differ (due to the application of different sample weights). I've updated the comment and added an explicit check for the expected differing checksums.